provider/cluster/kube/client.go

```diff
@@ -114,9 +118,7 @@ func (c *client) shouldExpose(expose *manifest.ServiceExpose) bool {
 }
 
 func (c *client) Deployments() ([]cluster.Deployment, error) {
-	// TODO: accept context as parameter
-	ctx := context.Background()
-	manifests, err := c.mc.AkashV1().Manifests(c.ns).List(ctx, metav1.ListOptions{})
+	manifests, err := c.mc.AkashV1().Manifests(c.ns).List(c.ctx, metav1.ListOptions{})
```

Comment:
Let's pass `context.Context` as an argument to these methods instead of saving it on `client`.

Reply:
:+1:

Reply:
:+1:

Reply:
@boz, so updating all of the functions to accept `ctx` results in the [Client interface](https://github.com/ovrclk/akash/blob/65bac28086e0e1a196897e39c05d7df675696cc7/provider/cluster/client.go#L22) needing to be updated. Do we want to go that far? That's a larger API change, and I'm less sure how far reaching it will be.

Reply:
Yeah let's definitely do it, but as a follow-on PR/issue.